packages/excalidraw/change.ts

```diff
@@ -0,0 +1,1342 @@
+import { ENV } from "./constants";
+import { LinearElementEditor } from "./element/linearElementEditor";
+import {
+  ElementUpdate,
+  mutateElement,
+  newElementWith,
+} from "./element/mutateElement";
+import {
+  getBoundTextElementId,
+  redrawTextBoundingBox,
+} from "./element/textElement";
+import { hasBoundTextElement, isBoundToContainer } from "./element/typeChecks";
+import {
+  BoundElement,
+  ExcalidrawLinearElement,
+  ExcalidrawTextElement,
+  NonDeleted,
+  Ordered,
+  OrderedExcalidrawElement,
+  SceneElementsMap,
+} from "./element/types";
+import { orderByFractionalIndex, syncMovedIndices } from "./fractionalIndex";
+import { getObservedAppState } from "./store";
+import {
+  AppState,
+  ObservedAppState,
+  ObservedElementsAppState,
+  ObservedStandaloneAppState,
+} from "./types";
+import { Mutable, SubtypeOf } from "./utility-types";
+import {
+  arrayToMap,
+  arrayToObject,
+  assertNever,
+  isShallowEqual,
+} from "./utils";
+
+/**
+ * Represents the difference between two objects of the same type.
+ *
+ * Both `deleted` and `inserted` partials represent the same set of added, removed or updated properties, where:
+ * - `deleted` is a set of all the previous (removed) values
+ * - `inserted` is a set of all the next (added, updated) values
```

Comment:
Either emphasise it's not always (for reference values only if we postprocess them), or make it generic-enough to be so that all the reference values are always postprocessed automatically.

Reply:
Won't be generic for now, reasoning https://github.com/excalidraw/excalidraw/pull/7348#discussion_r1521718445

Comment & description should be updates